### DISCUSSION TITLE
Improve nearby POI and favorites handling

### DISCUSSION THREAD

**Discussion by DolomitesTom** [devemux86/cruiser#267] [Q&A] [answered]
When tapping on a nearby favorite, normally the marker moves to this favorite. This sometimes or more often doesn't work well, my assumption is because of the high zoom in level, like in this example:

![20230802_194506](https://github.com/devemux86/cruiser/assets/121371629/692aefe4-7871-4748-bf78-292a847a17a1)

Maybe the minimax autozoom is the restriction.

Don't know if this possible, I guess it is a bit tricky to realize, but it would help a lot for an intuitive usage of nearby favorites or POIs:

When tapping on a nearby favorite or poi, any restriction should be ignored to move the cursor always to this position which should additionally be centered between the left, right and top navigation bar borders. Because sometimes it happens that the selected favorite is under the navigation or info bars, then you don't see it at all on the map. The zoom factor should be adapted so that the route and the favorite can be seen centered together. The favorite should be selected to show the name on the text bubble (like when manually selected) and the name should also be pronounced via voice output additionally. Otherwise it is often not clear witch favorite or POI is meant.
Same could be done for long press on the nearby favorite or POI, before asking for inserting into the route. Also here it could make a voice output additionally.

Hope I explained well what I mean 🙄🤔.

**Comment by devemux86** [comment]
@DolomitesTom thanks for the report.

Please post a route with the favorites or poi, also app settings, so I can reproduce the report.
And is it with simulation?

Current process, stops temporarily the auto center and centers the map on the tapped favorite  / poi.
Zoom remains the same. The map center is the center of navigation: screen center or bottom center.

The idea with the info bubble and speak the name is interesting. :slightly_smiling_face:

**Comment by DolomitesTom** [comment]
> And is it with simulation?

No, it was real navigation with DCT car. Therefore difficult to reproduce. 
I use option auto zoom mode 5, min = 13, max = 17, nearby favorites and POIs up to 15km and I can reproduce it with favorites more far away than the actually zoomed map areas

**Comment by devemux86** [comment]
Sorry, I still do not understand the report.

We navigate in landscape with "Movement direction" 2D or 3D. So the _map center is at bottom center of the screen_.
(image 1)

We tap a nearby POI / favorite: auto center is temporarily disabled and map (bottom) centers on the POI / favorite.
(image 2 - where the GPS location was)

![Screenshot_1691047793](https://github.com/devemux86/cruiser/assets/3484020/1a510535-524f-44fa-a376-0911c7a20738)
![Screenshot_1691047811](https://github.com/devemux86/cruiser/assets/3484020/aa62284e-331e-4ec1-b473-e3a8f1cb3639)

**Reply by devemux86** [reply]
If you mean when we tap a nearby POI / favorite to zoom out to a fixed zoom instead of using navigation's last zoom, we can do this.

**Reply by DolomitesTom** [reply]
Don't know how I can better explain it. Here for example I tapped on the 1,9km far away fuel station. The zoom is obviously too big to show this in the screen. It looks as if the GPS Position is always within the zoom. If the zoom doesn't change in this situation to a lower zoom level, then the fuel station is not visible.

![Screenshot_20230803-165215_Cruiser](https://github.com/devemux86/cruiser/assets/121371629/443867a6-9930-44ac-908b-d3cdfbd374a7)

This is the same route that I have sent you already.

**Reply by devemux86** [reply]
I will make the tap on nearby POI / favorites to map center at a fixed zoom.

However I still do not understand what you are doing and cannot reproduce it.
Each time it works like my above images, the map bottom centers at the POI.

Does anyone else have something to report when tap nearby POI / favorites?

**Reply by devemux86** [reply]
By the way the zoom has nothing to do with map center.
Map centers regardless of zoom on the selected marker.

It just centers with more zoomed in or out map, that's all.

**Reply by devemux86** [reply]
Also the same map center already happens all this time when we tap the top horizontal panel with the next waypoint name.

**Reply by devemux86** [reply]
@DolomitesTom if you can, please record a video so I can see what you mean.

**Comment by devemux86** [comment, marked as answer]
Improved in [Cruiser 3.0.29](https://play.google.com/store/apps/details?id=gr.talent.cruiser).

**Reply by DolomitesTom** [reply]
No crash any more, rest needs to be tested more intensivly.

Just one idea: the feature of nearby POI and favorite would also be a good feature for the pure follow Mode.

**Reply by devemux86** [reply]
> Just one idea: the feature of nearby POI and favorite would also be a good feature for the pure follow Mode.

It's in the plan, but probably needs a different workflow and presentation, so we can discuss that separately.
